internal/remote/remote_catalog.go

```diff
@@ -102,9 +124,96 @@ func (c *RemoteCatalog) Delete(
 ) error {
 	skrContext, err := c.skrContextFactory.Get(kyma)
 	if err != nil {
-		return fmt.Errorf("failed to get SkrContext for deleting RemoteCatalog: %w", err)
+		return fmt.Errorf("failed to get SKR context: %w", err)
 	}
 
 	moduleTemplates := c.moduleTemplateSyncAPIFactoryFn(c.kcpClient, skrContext.Client, &c.settings)
 	return moduleTemplates.DeleteAllManaged(ctx)
 }
+
+// GetModuleReleaseMetasToSync returns a list of ModuleReleaseMetas that should be synced to the SKR.
+// A ModuleReleaseMeta that is Beta or Internal is synced only if the Kyma is also Beta or Internal.
+func (c *RemoteCatalog) GetModuleReleaseMetasToSync(
+	ctx context.Context,
+	moduleReleaseMetas *[]v1beta2.ModuleReleaseMeta,
```

Comment:
I think that `moduleReleaseMetas` should be a return value instead of parameter. Currently it's feels very C-language style, where it's a common idiom to return values via pointer arguments - because C lacks multiple return types.
I know that this is how k8s client `.List()` works and honestly I don't understand why they did it this way, maybe something to avoid unnecessary list (slice) allocations. I think our use case is much simpler and we should not return values via parameters.